bundle-workflow/python/sign.py

```diff
@@ -1,37 +1,31 @@
 #!/usr/bin/env python
 
 import os
-import subprocess as sp
-import tempfile
-import uuid
 import argparse
-from manifests.bundle_manifest import BundleManifest
+from git.git_repository import GitRepository
+from manifests.build_manifest import BuildManifest
+from signing_workflow.signer import Signer
 
 parser = argparse.ArgumentParser(description = "Sign artifacts")
-parser.add_argument('manifest', type = argparse.FileType('r'), help="Manifest file.")
+parser.add_argument('manifest', type = argparse.FileType('r'), help="Path to local manifest file.")
+parser.add_argument('--component',nargs='?',help="Component name")
+parser.add_argument('--type',nargs='?',help="Artifact type")
 args = parser.parse_args()
 
-def sign(filename):
-    signatureFile = filename + ".asc"
-    signing_cmd = './opensearch-signer-client -i ' + filename + ' -o ' + signatureFile + ' -p pgp'
-    sp.run(signing_cmd.split(),cwd='opensearch-signer-client/src')
-    verify_cmd = 'gpg --verify-files ' + signatureFile
-    sp.run(verify_cmd.split(),cwd='opensearch-signer-client/src')
-    sp.run(['mv',signatureFile,'signed_artifacts/'])
+manifest = BuildManifest.from_file(args.manifest)
+basepath = os.path.dirname(os.path.abspath(args.manifest.name))
+git_repo = GitRepository("https://github.com/opensearch-project/opensearch-signer-client.git", "HEAD")
```

Comment:
nit: maybe `opensearch_signer_git_repo`, but also this feels like an implementation detail and maybe belongs inside `Signer`?